hazelcast/src/test/java/com/hazelcast/cluster/ConfigCheckTest.java

```diff
@@ -50,6 +61,86 @@ public void whenGroupNameDifferent_thenFalse() {
     }
 
     @Test
+    public void whenGroupPasswordDifferent_thenJoin() {
+        Config config1 = new Config();
+        config1.getGroupConfig().setName("foo");
+        config1.getGroupConfig().setPassword("Here");
+
+        Config config2 = new Config();
+        config2.getGroupConfig().setName("foo");
+        config2.getGroupConfig().setPassword("There");
+
+        ConfigCheck configCheck1 = new ConfigCheck(config1, "joiner");
+        ConfigCheck configCheck2 = new ConfigCheck(config2, "joiner");
+
+        assertIsCompatibleTrue(configCheck1, configCheck2);
+    }
+
+    @Test
+    public void testGroupPasswordLeak_whenVersionUnknow() {
+        final Config config = new Config();
+        config.getNetworkConfig().getJoin().getMulticastConfig()
+               .setEnabled(true).setMulticastTimeoutSeconds(3);
+        config.getNetworkConfig().getJoin().getTcpIpConfig().setEnabled(false);
+
+        final AtomicBoolean leaked = new AtomicBoolean(false);
+
+        ObjectDataOutput odo = mock(ObjectDataOutput.class);
+
+        try {
```

Comment:
I think you can remove all these try/catches. Just let the exception propagate; the test will fails as well.